playerSDK/src/main/java/com/kaltura/playersdk/players/KChromeCastPlayer.java

```diff
@@ -132,69 +132,77 @@ private void stopTimer() {
     }
 
     public void play() {
-        LOGD(TAG, "Start PLAY");
-        if (isEnded) {
-            load(0);
-            isEnded = false;
-            stopTimer();
-            startTimer();
-            updateState(State.Playing);
-            return;
-        }
+        if (hasMediaSession()) {
```

Comment:
Instead of wrapping the "positive" code with an if, exit the function on the negative case. Then there are less changes in the code, the changes are easy to see, and the lines are shorter.
Same for the other functions below.